transcendence-app/src/user/user.service.ts

```diff
@@ -14,6 +18,20 @@ export class UserService {
   retrieveUserWithId(id: string) {
     return this.users.find((user) => user.id === id) || null;
   }
+  retrieveUsers(queryDto: UsersPaginationQueryDto) {
+    if (queryDto.limit || queryDto.offset) {
+      let ret: string[] = [];
+      queryDto.limit = queryDto.limit
+        ? queryDto.limit
+        : MAX_USER_ENTRIES_PER_PAGE;
+      queryDto.offset = queryDto.offset ? queryDto.offset : 0;
+      ret = this.users
+        .filter((user, i) => i >= queryDto.offset && i < queryDto.limit)
```

Comment:
A query like this:
`http://127.0.0.1:3000/api/v1/users?limit=20&offset=19&sort=true`

Shows only one username: the 19th one. How do we show any username at index past 19? Can you show me a query that does that? If I set limit higher than 20 it returns 400 Bad Request. From my understanding, a query like the above should still show 20 users, starting with the one in the position 19th. Am I wrong?

Reply:
`i <= queryDto.limit` should fix that?

Reply:
No, as I established a MAX() in the Dto, No limit above 20 will be considered. I'll fix it.

Reply:
I've changed it to return always user. Without parameters it returns the maximum number of users established, and with queryparams, what has been selected. 
http://127.0.0.1:3000/api/v1/users?limit=20&offset=19&sort=true would return 20 users, from the 19th.